### PR TITLE
feat(sdk-lib-mpc): add proof of paillier correctness & refactor appendChallenge (1/4)

### PR DESCRIPTION
**Pull request by zahin-mohammad**
Note: this is a refactor of #3502

# Please be aware of the following when making your pull request:

## Description

- introduces the primitives for pallier proofs from the PR listed above
- refactors appendChallenge to be simpler

## Issue Number

TICKET: BG-78338

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code compiles correctly for both Node and Browser environments
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [x] The ticket or github issue was included in the commit message as a reference
- [x] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: modules/sdk-lib-mpc/test/unit/tss/ecdsa/pallierproof.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      }
+    });
+
+    it('should generate 128 challenges', async function () {
+      const n = await randomBigInt(minModulusBitLength);
+      assert.strictEqual((await generateP(BigInt(n))).length, 128);
+    });
+  });
+
+  // describe('prove', function () {});
```
Comment (zahin-mohammad) [inline modules/sdk-lib-mpc/test/unit/tss/ecdsa/pallierproof.ts:28]: Will be done in a follow up PR (2/2)

File: modules/sdk-lib-mpc/src/tss/ecdsa/pallierproof.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
+import { bitLength } from 'bigint-crypto-utils';
```
Comment (zahin-mohammad) [inline modules/sdk-lib-mpc/src/tss/ecdsa/pallierproof.ts:1]: Copied from Johns PR

File: modules/sdk-lib-mpc/src/util.ts [hunk 3]
```diff
@@ -112,11 +112,14 @@ export async function randomCoPrimeTo(x: bigint): Promise<bigint> {
...
 /**
- * Generate a random coprime less than x with the same bit depth.
+ * Generate a random positive integer coprime less than x with the same bit depth.
  * @param x
  * @returns {Promise<bigint>}
  */
-export async function randomCoPrimeLessThan(x: bigint): Promise<bigint> {
+export async function randomPositiveCoPrimeLessThan(x: bigint): Promise<bigint> {
+  if (x < BigInt(2)) {
+    throw new Error('x must be larger than 2');
```
Comment (islamaminBitGo) [inline modules/sdk-lib-mpc/src/util.ts:121]: or equal to 2

File: modules/sdk-lib-mpc/src/tss/ecdsa/index.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
 export * as EcdsaTypes from './types';
 export * as EcdsaRangeProof from './rangeproof';
 export * as EcdsaPallierProof from './pallierproof';
+
+export const minModulusBitLength = 3072;
```
Comment (islamaminBitGo) [inline modules/sdk-lib-mpc/src/tss/ecdsa/index.ts:5]: nit: will benefit from changing all occurrences of 3072 to minModulusBitLength, some places are still not changed.

**Review by islamaminBitGo — APPROVED** [review]

**Review by margueriteblair — APPROVED** [review]
altcoin changes look good to me

**Review by rlutzbitgo — APPROVED** [review]

**Pull request merged**